enhanced/enhanced.go

```diff
@@ -209,33 +209,31 @@ func (e *Exporter) collectValue(ch chan<- prometheus.Metric, instance config.Ins
 			}
 			l.TakeOldest(t)
 		default:
-			return fmt.Errorf("unsupported key '%s' when collecting value", key)
+			log.Error(fmt.Errorf("unsupported key '%s' when collecting value", key))
 		}
 	default:
-		return fmt.Errorf("unsupported value type '%T' for key '%s' when collecting value", value, key)
+		log.Error(fmt.Errorf("unsupported value type '%T' for key '%s' when collecting value", value, key))
 	}
-
 	return nil
 }
 
-func (e *Exporter) collectMapValue(ch chan<- prometheus.Metric, instance config.Instance, key string, value map[string]interface{}, l *latency.Latency, extraLabelsValues ...string) error {
+func (e *Exporter) collectMapValue(ch chan<- prometheus.Metric, instance *config.Instance, key string, value map[string]interface{}, l *latency.Latency, extraLabelsValues ...string) error {
 	for metricName, v := range value {
 		metricValue, ok := v.(float64)
 		if !ok {
-			return fmt.Errorf("%s: wrong value type for metric %s: %T", key, metricName, value)
+			log.Error(fmt.Errorf("%s: wrong value type for metric %s: %T", key, metricName, value))
 		}
```

Comment:
`continue` after this `log.Error` and please verify if other `return` to `log.Error` changes don't require early stop as well.

Reply:
As far as I can see and made tests, we are safe on this. And actually, it helps to debug issues with metrics processing.

Reply:
If you don't add `continue` after `log.Error` here it will send `0` value for the metric. I don't see how this is desired.

Reply:
yeap, fixed it